src/inventory_ui.cpp

```diff
@@ -887,8 +896,10 @@ item_location game::inv_map_splice(
             auto &current_stack = stacks.back();
             current_stack.reserve( items.size() );
 
+            bool found_item = false;
```

Comment:
A `found_foo` variable is a sign that the loop should go into a separate function. There you could just leave the function *and* the loop via `return` when the item has been found, or (if not found) create the new item and add it.